src/components/list/ItemsList.tsx

```diff
@@ -69,22 +84,29 @@ export default class ItemsList<T extends ListItem> extends React.Component<Props
     );
   }
 
-  private onSelectItem(item: T) {
-    const { selectedIds } = this.state;
+  private onSelectItem(item: T): void {
+    const { selectedItems } = this.state;
+    const { onSelect, select } = this.props;
+    const callback = onSelect ? () => onSelect([...this.state.selectedItems.values()]) : null;
     const id = item.id;
-    // If the item is already selected, unselect it
-    if (selectedIds.has(id as string)) {
-      const newSelectedIds = new Set(selectedIds);
-      newSelectedIds.delete(id as string);
-      this.setState({ ...this.state, selectedIds: newSelectedIds });
+    // If the item is already selected
+    if (selectedItems.has(id)) {
+      // If the item is not the only one selected, unselect it
+      if (selectedItems.size > 1) {
+        selectedItems.delete(id);
+        this.setState({ selectedItems }, callback);
+      }
       // Else, add the item to the selected Ids
     } else {
-      switch (this.props.select) {
+      switch (select) {
         case ItemsListTypes.MULTI:
-          this.setState({ ...this.state, selectedIds: new Set(selectedIds).add(item.id as string) });
+          this.setState({ selectedItems: selectedItems.set(id, item) }, callback);
```

Comment:
No several statements in one line.
I would expect:

```
selectedItems.set(id, item);
callback();
```
The update of the list should be done by the parent I think because it has to rerender the selected UI element.

Reply:
The ItemsList component is the one who call the renders method for each item and passes a boolean to specify if that item should be selected